extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/JaxRsAnnotationScannerTest.java

```diff
@@ -61,6 +62,31 @@ void testJakartaHiddenOperationNotPresent() throws IOException, JSONException {
         testHiddenOperationNotPresent(indexer.complete());
     }
 
+    /**
+     * Verify that no false positive failures/warnings for duplicate operation-IDs are generated when both
+     * {@code javax} and {@code jakarta} annotations are present.
+     */
+    @Test
+    void testMixedJakartaAndJavaxAnnotations() throws IOException {
+        // Needed a resource implementation class and an interface to trigger the duplicate operation-ID
+        // false positives.
+        Indexer indexer = new Indexer();
+        index(indexer, "test/io/smallrye/openapi/runtime/scanner/resources/mixed/RestResourceImpl.class");
+        index(indexer, "test/io/smallrye/openapi/runtime/scanner/resources/mixed/RestInterface.class");
+        Index index = indexer.complete();
+
+        Map<String, String> cfg = new HashMap<>();
+        // fail hard to trigger the failure in this test
+        cfg.put(OpenApiConstants.DUPLICATE_OPERATION_ID_BEHAVIOR, OpenApiConfig.DuplicateOperationIdBehavior.FAIL.toString());
+        // method-strategy needed for this test
+        cfg.put(OpenApiConstants.OPERATION_ID_STRAGEGY, OpenApiConfig.OperationIdStrategy.METHOD.toString());
+        OpenApiConfig config = dynamicConfig(cfg);
+
+        OpenApiAnnotationScanner s = new OpenApiAnnotationScanner(config, index);
+        OpenAPI result = s.scan("JAX-RS");
```

Comment:
Maybe something like this will make explicit that nothing will be thrown even with the `FAIL` DuplicateOperationIdBehavior.
```suggestion
        OpenAPI result = assertDoesNotThrow(() -> s.scan("JAX-RS"));
```